pkg/registry/populator.go

```diff
@@ -119,28 +167,38 @@ func (i *DirectoryPopulator) loadManifests(imagesToAdd []*ImageInput, mode Mode)
 		// same package linearly.
 		var err error
 		var validImagesToAdd []*ImageInput
+
+		for pkg := range i.overwriteDirMap {
+			err := i.loader.RemovePackage(pkg)
```

Comment:
I think we want the remove / add to happen in a single transaction. 

It would also be nice to not need to thread through the "overwriteImages" separately everywhere. is there a way we can we push the `overwrite` option down to `loadManifestReplaces` (or call `loadManifestsReplaceOverwrite` if overwrite is set, etc), so that we only need to worry about overwriting when we try to insert and it fails because it already exists?

Reply:
we can but that requires exposing the image registry package to the registry package which today can't easily be done because of cyclic imports...

Reply:
I think you could avoid importing the image/registry package with a `ReplacePackage(pkg string, bundle ...Bundle)` method on the loader -- still getting a single transaction at the DB level. 

However, I do think that at this point, we should prioritize getting something working and merged over long-term support; partly b/c the declarative index config will supersede this work in the near future.